src/main/kotlin/com/supergrecko/questionbot/services/QuestionService.kt

```diff
@@ -0,0 +1,26 @@
+package com.supergrecko.questionbot.services
+
+import com.supergrecko.questionbot.dataclasses.BotConfig
+import com.supergrecko.questionbot.dataclasses.GuildConfig
+import me.aberrantfox.kjdautils.api.annotation.Service
+import me.aberrantfox.kjdautils.discord.Discord
+import me.aberrantfox.kjdautils.internal.di.PersistenceService
+
+private lateinit var guilds: MutableList<GuildConfig>
+
+/**
+ * Get a guild by its id
+ *
+ * @param id the guild snowflake
+ */
+fun getGuild(id: String?): GuildConfig {
+    return guilds.first { it.guild == id }
+}
+
+@Service
+class QuestionService(val config: BotConfig, private val discord: Discord, private val store: PersistenceService) {
+    init {
+        // Hacky way to extract guilds from service
```

Comment:
I'd probably want a less hacky way to extract the guilds into the parent scope, but this will do for now.